src/update-feeds-loops/contracts.ts

```diff
@@ -10,25 +10,23 @@ import { zip } from 'lodash';
 import type { AirnodeAddress, TemplateId } from '../types';
 import { decodeDapiName, deriveBeaconId } from '../utils';
 
-export const createProvider = async (url: string) => {
+export const createProvider = async (chainId: string, chainAlias: string, providerUrl: string) => {
   // Create a static network provider (bound to a specific network) to avoid the overhead of fetching the chain ID
   // periodically.
-  const provider = new ethers.JsonRpcProvider(url, undefined, {
-    staticNetwork: true,
-  });
-
-  // Unfortunately, ethers internally detects the network and when it fails it retries indefinitely (and logs and
-  // annoying message). See:
+  //
+  // Specifying the chain statically also avoids the code path that can cause infinite retries when the RPC is down.
+  // See:
   // https://github.com/ethers-io/ethers.js/blob/ad5f1c5fc7b73cfb20d9012b669e9dcb9122d244/src.ts/providers/provider-jsonrpc.ts#L762
   //
-  // This is extremely unwanted behaviour, because if there is a broken RPC this will drain resources. Moreover, in
-  // Airseeker we initialize provider in every cycle. The workaround is to detect the network manually to ensure the RPC
-  // is healthy. If this is not the case, we destroy the provider stopping it from draining any resources.
-  const goDetectNetwork = await go(async () => provider._detectNetwork());
-  if (!goDetectNetwork.success) {
-    provider.destroy();
-    return null;
-  }
+  // This is extremely unwanted behaviour, because Airseeker initializes provider in every cycle.
+  const network = new ethers.Network(chainAlias, chainId);
+  const provider = new ethers.JsonRpcProvider(providerUrl, network, {
```

Comment:
This time, I made sure the problematic code path is not called at all. This is because the provider is initialized with a `Network` object, which makes [this condition](https://github.com/ethers-io/ethers.js/blob/ad5f1c5fc7b73cfb20d9012b669e9dcb9122d244/src.ts/providers/provider-jsonrpc.ts#L762) return false and the while loop never runs.